lib/execution_engine2/utils/auth.py

```diff
@@ -0,0 +1,94 @@
+"""
+A utility class for fetching roles for a token and caching them.
+This makes use of the TokenCache in the built-in service auth client.
+
+This is an SDK-based service, so auth token validation is handled through
+the Server. This just takes an auth token and discerns whether there's
+a valid EE2 admin role attached to the user.
+"""
+from typing import List, Set
+from installed_clients.authclient import (
+    TokenCache
+)
+import requests
+from json import JSONDecodeError
+from ..exceptions import AuthError
+
+IS_ADMIN = "EE2_ADMIN"
+NOT_ADMIN = "NOT_ADMIN"
+
+_admin_cache = TokenCache()
+
+
+class AuthUtil(object):
```

Comment:
We probably don't need to be python2 compatible, so we don't have to import from object

Reply:
Yup, thanks for catching that. Just a habit.